pandas/tests/indexes/test_base.py

```diff
@@ -1270,6 +1270,24 @@ def test_sortlevel_na_position(self):
         expected = Index([np.nan, 1])
         tm.assert_index_equal(result, expected)
 
+    def test_index_diff(self):
+        # GH#19708
+        idx = Index([10, 20, 30, 40, 50])
+        diffs = idx.diff()
+        diffs_period2 = idx.diff(periods=2)
```

Comment:
Could you use `pytest.mark.parameterize` to test different periods?

Reply:
Would you like pytest.mark.parametrize for the round_test as well?

Reply:
Yes please